A80-pid.md

```diff
@@ -0,0 +1,290 @@
+
+The `pid` balancer takes a different approach: instead of deterministically calculating weights based on a backend metric, it continuously adjusts weights at runtime. It utilizes a feedback loop based on the backend CPU metric to determine the direction and magnitude of every weight update.
+
+
+### Related Proposals:
+* [gRFC A51: Custom Backend Metrics Support][A51]
+* [gRFC A58: `weighted_round_robin` LB policy][A58]
+* [gRFC A68: Random subsetting with rendezvous hashing LB policy.][A68]
+
+## Proposal
+
+Introduce a new LB policy `pid`. This policy implements client-side load balancing with direct load reports from backends. It utilizes a feedback loop with a PID controller to dynamically adjust the weights. The policy is otherwise largely a superset of the existing policy `weighted_round_robin`.
+
+### LB Policy Config and Parameters
+
+The `pid` LB policy config will be as follows.
+
+```textproto
+message LoadBalancingConfig {
+  oneof policy {
+    PIDLbConfig pid = 20 [json_name = "pid"];
+  }
+}
+
+message PIDLbConfig {
+  // Configuration for the WRR load balancer as defined in [gRFC A58][A58].
+  // The PID balancer is an extension of WRR and all settings applicable to WRR also apply to PID identically.
+  WeightedRoundRobinLbConfig wrr_config = 1;
+
+  // Threshold beyond which the balancer starts considering the ErrorUtilizationPenalty.
+  // This helps avoid oscillations in cases where the server experiences a very high and spiky error rate.
+  // We avoid eliminating the error_utilization_penalty entirely to prevent redirecting all traffic to an instance 
+  // that has low CPU usage but rejects all requests. Default is 0.5.
+  google.protobuf.FloatValue error_utilization_threshold = 2;
+
+  // Controls the convergence speed of the PID controller. Higher values accelerate convergence but may induce oscillations,
+  // especially if server load changes more rapidly than the PID controller can adjust. Oscillations might also occur due to
+  // significant delays in load report propagation or extremely spiky server load. To mitigate spiky loads, server owners should
+  // employ a moving average to smooth the load reporting. Default is 0.1.
+  google.protobuf.FloatValue proportional_gain = 2;
+
+  // Adjusts the smoothness of the PID controller convergence. Higher values enhance smoothness but can decelerate convergence.
+  // Default is 1.
+  google.protobuf.FloatValue derivative_gain = 4;
+
+  // Maximum allowable weight. Weights proposed by the PID controller exceeding this value will be capped.
+  // This prevents infinite weight growth, which could occur if only a subset of clients uses PID and increasing weights
+  // no longer effectively corrects the imbalance. Default is 10.
+  google.protobuf.FloatValue max_weight = 5;
```

Comment:
This is a very good point,  I verified that the drift actually happens in practice. It is not causing issues and we didn't notice this before, since weights are re-centered around 1 after every client redeployment and, as you said, drift stops once a significant portion of the clients reach the limit. 

Anyway, I'll update gRFC to include your proposal.